pact/Pact/Core/IR/Term.hs

```diff
@@ -494,29 +510,76 @@ termInfo f = \case
   CapabilityForm cf i -> CapabilityForm cf <$> f i
   ObjectLit m i -> ObjectLit m <$> f i
 
+traverseTerm
```

Comment:
Given the confusions we had with missing expected subterms, we should probably add a `-- TODO here`:

```
-- TODO: add several tests to confirm we extracted the expected subterms from various complex nested terms`.
```